lib/i18n/config.rb

```diff
@@ -110,12 +110,14 @@ def missing_interpolation_argument_handler=(exception_handler)
     # register translation files like this:
     #   I18n.load_path << 'path/to/locale/en.yml'
     def load_path
+      reset_available_locales_set
       @@load_path ||= []
```

Comment:
`load_path=` must be defined as a method, and in case of resetting it should trigger reset of available locales.